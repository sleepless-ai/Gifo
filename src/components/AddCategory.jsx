import { useState } from 'react';

export const AddCategory = ({onNewCategory}) => {
  
  const [ inputValue, setInputValue ] = useState('')
  
  const [ isError, setIsError ] = useState(false)
  
  const onInputChange = (e) => {
    setInputValue(e.target.value)
  }
  
  const onSubmit = (e) => {
    e.preventDefault();
    
    if (inputValue.trim().length <= 1) {
      setIsError(true)
      setInputValue('')
      return
    };
    
    onNewCategory(inputValue.trim())
    setInputValue("")
  }
  
  return(
    <form onSubmit={onSubmit}>
      {
        isError && 
        <h3 className="text-red-400">Tienes que agregar mas de un caracter</h3>
      }
      <input
        className="p-2"
        placeholder="Buscar gif"
        value={inputValue}
        onChange={onInputChange}
      />
      <button className="p-2 border border-black rounded">Agregar</button>
    </form>
    )
}