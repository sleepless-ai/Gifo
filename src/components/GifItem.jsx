export const GifItem= ({title, url, id}) => {
  return (
    <>
      <h3>{title}</h3>
      <img src={ url } alt={ title }/>
    </>
    )
}