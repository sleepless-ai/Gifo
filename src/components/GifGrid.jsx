import { useState, useEffect } from 'react'

import { getGifs } from '../helpers/getGifs'
import { GifItem } from './GifItem'

export const GifGrid = ({category}) => {
  
  const [ images, setImages ] = useState([])
  
  const getImages = async() => {
    const newImages = await getGifs( category )
    setImages(newImages)
  }
  
  useEffect(() => {
    getImages()
    return
  }, [])
  
  return (
    <div className="py-4">
      <h3 className="font-bold text-xl">{category}</h3>
      <div className="py-4 flex flex-col gap-4">
      {
        images.map((images) => {
          return (
              <GifItem
                key={images.id}
                {...images}
              />
            )
        })
      }
      </div>
    </div>
    )
}