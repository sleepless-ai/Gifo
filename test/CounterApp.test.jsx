import { render, screen } from "@testing-library/react"
import { CounterApp } from "../src/components/CounterApp"

describe("Pruebas en <App />",() => {
  test("esta prueba no falla", () => {
    const { container } = render(<CounterApp/>)
    expect( container ).toMatchSnapshot();
  })
})