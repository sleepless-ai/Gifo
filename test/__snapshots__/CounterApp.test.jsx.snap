// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <App /> esta prueba no falla 1`] = `
<div>
  <div>
    Holi
  </div>
</div>
`;
